src/com/google/enterprise/adaptor/filenet/DocumentTraverser.java

```diff
@@ -393,6 +405,87 @@ private void setMetadata(Document document, Response response) {
     logger.log(Level.FINEST, "Property names: {0}", names);
   }
 
+  private List<String> getPropertyValues(Property prop) {
+    ArrayList<String> list = new ArrayList<>();
+    if (prop instanceof PropertyString) {
+      getValue(prop.getStringValue(), list);
+    } else if (prop instanceof PropertyStringList) {
+      getListValue(prop.getStringListValue(), list);
+    } else if (prop instanceof PropertyBoolean) {
+      getValue(prop.getBooleanValue(), list);
+    } else if (prop instanceof PropertyBooleanList) {
+      getListValue(prop.getBooleanListValue(), list);
+    } else if (prop instanceof PropertyDateTime) {
+      getDateValue(prop.getDateTimeValue(), list);
+    } else if (prop instanceof PropertyDateTimeList) {
+      getDateListValue(prop.getDateTimeListValue(), list);
+    } else if (prop instanceof PropertyFloat64) {
+      getValue(prop.getFloat64Value(), list);
+    } else if (prop instanceof PropertyFloat64List) {
+      getListValue(prop.getFloat64ListValue(), list);
+    } else if (prop instanceof PropertyInteger32) {
+      getValue(prop.getInteger32Value(), list);
+    } else if (prop instanceof PropertyInteger32List) {
+      getListValue(prop.getInteger32ListValue(), list);
+    } else if (prop instanceof PropertyId) {
+      getGuidValue(prop.getIdValue(), list);
+    } else if (prop instanceof PropertyIdList) {
+      getGuidListValue(prop.getIdListValue(), list);
+    } else {
+      logger.log(Level.FINEST, "Property type is not supported: {0}",
+          prop.getClass().getName());
+    }
+    return list;
+  }
+
+  private void getValue(Object val, List<String> valuesList) {
+    if (val != null) {
+      valuesList.add(val.toString());
+    }
+  }
+
+  private void getListValue(List<?> values, List<String> valuesList) {
+    for (Object val : values) {
+      if (val != null) {
+        valuesList.add(val.toString());
+      }
+    }
+  }
+
+  private void getGuidValue(Id val, List<String> valuesList) {
+    if (val != null) {
+      String id = val.toString();
+      valuesList.add(id.substring(1, id.length() - 1));
+    }
+  }
+
+  private void getGuidListValue(List<?> values, List<String> valuesList) {
+    for (Object val : values) {
+      if (val != null) {
+        String id = val.toString();
+        valuesList.add(id.substring(1, id.length() - 1));
+      }
+    }
+  }
+
+  private void getDateValue(Date val, List<String> valuesList) {
+    if (val != null) {
+      Calendar cal = Calendar.getInstance();
+      cal.setTime(val);
+      valuesList.add(/*TODO: ISO 8601*/val.toString());
```

Comment:
I don't like to edit moved code blocks like this, plus I have this lumped into the TODOs task in my mind.